### DISCUSSION TITLE
Installation issues

### DISCUSSION THREAD

**Discussion by matheusdiogenesandrade** [cvanaret/Uno#8] [Q&A] [answered]
Just downloaded the package, and followed steps:

```sh
mkdir build
cd build
cmake -Dma57=/home/user/hsl/hsl_ma57-5.2.0/compiled/lib/libhsl_ma57 -Damplsolver=/home/user/ampl/amplapi/lib/libampl ..
make
```

And obtained the following message

```sh
...
In file included from ~/Uno/uno/optimization/ModelFactory.hpp:10,
                 from ~/Uno/uno/optimization/ModelFactory.cpp:4:
~/Uno/uno/interfaces/AMPL/AMPLModel.hpp:13:10: fatal error: asl_pfgh.h: No existe el archivo o el directorio
   13 | #include "asl_pfgh.h"
      |          ^~~~~~~~~~~~
compilation terminated.
make[2]: *** [CMakeFiles/uno.dir/build.make:482: CMakeFiles/uno.dir/uno/optimization/ModelFactory.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:85: CMakeFiles/uno.dir/all] Error 2
make: *** [Makefile:91: all] Error 2
```

I checked the folder `/home/user/ampl/solvers` to see if the desired file is available.

```sh
ls ~/ampl/solvers/asl_pfg*

/home/user/ampl/solvers/asl_pfg.h  /home/user/ampl/solvers/asl_pfgh.h
```

Should I link the ASL headers all well?

Thanks and regards.

**Comment by matheusdiogenesandrade** [comment]
The same happens when using the flag `-DCMAKE_BUILD_TYPE=Release`.

**Comment by cvanaret** [comment, marked as answer]
Hi Matheus,
You need to link the ASL headers only, not the actual libampl. So it should look something like:

`cmake -Dma57=/home/user/hsl/hsl_ma57-5.2.0/compiled/lib/libhsl_ma57 -Damplsolver=/home/user/ampl/solvers ..`

Hope that helps! I look forward to your feedback on Uno.

**Reply by matheusdiogenesandrade** [reply]
Thank you so much.
